core/src/main/java/org/infinispan/filter/KeyFilter.java

```diff
@@ -10,8 +10,10 @@
  *
  * @author Manik Surtani
  * @since 6.0
+ * @deprecated This will be replaced by {@link java.util.function.Predicate} in the future
  */
 @ThreadSafe
+@Deprecated
```

Comment:
Why don't you let this already extend Predicate? That way you could safely change the method parameters to Predicate because all KeyFilters passed in would be Predicates...

Reply:
Sure, I am not sure why I didn't do this before 👍